rl4co/models/zoo/common/nonautoregressive/decoder.py

```diff
@@ -156,33 +156,27 @@ def forward(
 
         # Main decoding: loop until all sequences are done
         while not td["done"].all():
-            log_p, mask = self._get_log_p(td, heatmaps_logp, num_starts)
-            td = decode_strategy.step(log_p, mask, td)
+            logits, mask = self._get_logits(td, heatmaps_logits, num_starts)
+            td = decode_strategy.step(logits, mask, td)
```

Comment:
Not 100% sure about this, since you use the sigmoid actually we may not need to  normalize the "logits". Or do we? @Furffico

Reply:
> Not 100% sure about this, since you use the sigmoid actually we may not need to normalize the "logits". Or do we? @Furffico

That's correct, we don't need to normalize the logits in this version. I tested and it worked fine.

Reply:
I see - but then isn't it better to use `log_softmax` on the heatmap rows rather than normalizing each element with sigmoid? 

Anyways this version should give the same solution in terms of sampling and greedy but it also supports the new features. So if you agree we could leave it as is?